ophyd/controls/device.py

```diff
@@ -233,7 +240,77 @@ def __new__(cls, name, bases, clsdict):
         return clsobj
 
 
-class OphydDevice(OphydObject, metaclass=ComponentMeta):
+
+# These stub 'Interface' classes are the apex of the mro heirarchy for
+# their respective methods. They make it safe to use multiple inheritance
+# with the base classes given in any order.
+
+
+class BlueskyInterface:
+    """Classes that inherit from this can safely customize the
+    these methods without breaking mro."""
+    def __init__(self, *args, **kwargs):
+        # Subclasses can populate this with (signal, value) pairs, to be
+        # set by stage() and restored back by unstage().
+        self.stage_sigs = list()
+        self._staged = False
+        super().__init__(*args, **kwargs)
+
+    def trigger(self):
+        pass
+
+    def read(self):
+        return {}
+
+    def describe(self):
+        return {}
+
+    def stage(self):
+        "Prepare the device to be triggered."
+        # Read and stage current values, to be restored by unstage()
+        self._original_vals = {sig: sig.get() for sig, _ in self.stage_sigs}
+
+        # Apply settings.
+        self._staged = True
+        for sig, val in self.stage_sigs:
+            ttime.sleep(0.1)
+            set_and_wait(sig, val)
+
+        # Call stage() on child devices (including, notably, plugins).
+        for signal_name in self.signal_names:
+            signal = getattr(self, signal_name)
+            if hasattr(signal, 'stage'):
+                signal.stage()
+
+    def unstage(self):
+        """
+        Restore the device to 'standby'.
+
+        Multiple calls (without a new call to 'stage') have no effect.
+        """
+        if not self._staged:
```

Comment:
Done. Also, reconsidering stage/unstage, I think repeated calls to `stage` should raise. Our options are:

1. Just stage again -- thus erasing the "original values", which is bad.
2. Fail loudly.
3. Fail silently.

I don't want to have to wonder whether it is passing silently (debug logging notwithstanding). Repeated calls to unstage are harmless, though, so that one can pass silently with a debug message, as it was.